source/connection-monitoring-and-pooling/tests/unified/network-error.json

```diff
@@ -0,0 +1,464 @@
+            }
+          ]
+        }
+      ],
+      "outcome": [
+        {
+          "collectionName": "cmap-network-error",
+          "databaseName": "cmap-tests",
+          "documents": [
+            {
+              "_id": 1
+            },
+            {
+              "_id": 2
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "description": "Pool properly handles network error during checkout",
+      "runOnRequirements": [
+        {
+          "minServerVersion": "4.4",
+          "serverless": "forbid",
+          "topologies": [
+            "single",
+            "sharded"
+          ]
+        }
+      ],
+      "operations": [
+        {
+          "name": "createEntities",
+          "object": "testRunner",
+          "arguments": {
+            "entities": [
+              {
+                "client": {
+                  "id": "client",
+                  "useMultipleMongoses": false,
+                  "observeEvents": [
+                    "connectionCheckOutStartedEvent",
+                    "connectionCheckedOutEvent",
+                    "connectionCheckOutFailedEvent",
+                    "connectionCheckedInEvent",
+                    "connectionClosedEvent",
+                    "poolClearedEvent",
+                    "connectionCreatedEvent",
+                    "connectionReadyEvent"
+                  ],
+                  "uriOptions": {
+                    "retryWrites": false,
+                    "retryReads": false,
+                    "appName": "cmapNetworkErrorCheckOutTest"
+                  }
+                }
+              },
+              {
+                "database": {
+                  "id": "database",
+                  "client": "client",
+                  "databaseName": "cmap-tests"
+                }
+              },
+              {
+                "collection": {
+                  "id": "collection",
+                  "database": "database",
+                  "collectionName": "cmap-network-error"
+                }
+              }
+            ]
+          }
+        },
+        {
+          "name": "waitForEvent",
+          "object": "testRunner",
+          "arguments": {
+            "client": "client",
+            "event": {
+              "poolReadyEvent": {}
+            },
+            "count": 1
+          }
+        },
+        {
+          "name": "failPoint",
+          "object": "testRunner",
+          "arguments": {
+            "client": "setupClient",
+            "failPoint": {
+              "configureFailPoint": "failCommand",
+              "mode": {
+                "times": 50
+              },
+              "data": {
+                "failCommands": [
+                  "isMaster",
+                  "hello"
+                ],
+                "closeConnection": true,
+                "appName": "cmapNetworkErrorCheckOutTest"
+              }
+            }
+          }
+        },
+        {
+          "name": "find",
```

Comment:
I think there might be a race here:
Pool is paused before Find happens, and then we don't get `connectionCreated`/`connectionClosed `event.

Reply:
good catch, i added a really high heartbeatFrequency to this test so that the failpoint wont be triggered by the monitors.